pkg/apis/ceph.rook.io/v1/pool.go

```diff
@@ -53,34 +53,47 @@ func (p *ReplicatedSpec) IsTargetRatioEnabled() bool {
 func (p *CephBlockPool) ValidateCreate() error {
 	logger.Infof("validate create cephblockpool %v", p)
 
-	err := validatePoolSpec(p.Spec.ToNamedPoolSpec())
+	err := ValidateCephBlockPool(&p.Spec)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+// ValidateCephBlockPool validates specifically a CephBlockPool's spec (not just any NamedPoolSpec)
+func ValidateCephBlockPool(s *NamedBlockPoolSpec) error {
+	name := s.Name
+	if name == "device_health_metrics" || name == ".mgr" || name == ".nfs" {
+		if s.IsErasureCoded() {
+			return errors.Errorf("invalid CephBlockPool spec: ceph built-in pool %q cannot be erasure coded", name)
+		}
+	}
+
+	return validatePoolSpec(s.ToNamedPoolSpec())
+}
```

Comment:
I figured that we would only want to check the built-in pool names for CephBlockPools and not for named pools that might appear on file/object CRs.